libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -194,24 +195,111 @@ impl<'a> LiftAstToDml<'a> {
+            if supports_native_types {
+                let (connector_string, connector) = match self.source {
+                    Some(source) => (&source.active_provider, &source.active_connector),
+                    None => panic!(""),
+                };
+                let mut connectors: HashMap<&str, Box<dyn Connector>> = HashMap::new();
+                connectors.insert("postgresql", Box::new(SqlDatamodelConnectors::postgres()));
+                connectors.insert("mysql", Box::new(SqlDatamodelConnectors::mysql()));
+                connectors.insert("sqlite", Box::new(SqlDatamodelConnectors::sqlite()));
+                connectors.insert("mssql", Box::new(SqlDatamodelConnectors::mssql()));
+
+                let prefix = format!("{}{}", datasource_name, ".");
+
+                let type_specifications = ast_field
+                    .directives
+                    .iter()
+                    .filter(|dir| dir.name.name.starts_with(&prefix))
+                    .collect_vec();
+
+                if type_specifications.len() > 1 {
+                    return Err(DatamodelError::new_duplicate_directive_error(
+                        &prefix,
+                        type_specifications.first().unwrap().span,
+                    ));
+                }
+
+                let name = type_specifications
+                    .first()
+                    .map(|dir| dir.name.name.trim_start_matches(&prefix));
+
+                // convert arguments to u32 if possible
+                let number_args = type_specifications.first().map(|dir| dir.arguments.clone());
+                let args = if let Some(number) = number_args {
+                    let p = number
+                        .iter()
+                        .map(|arg| ValueValidator::new(&arg.value).as_int())
+                        .collect_vec();
+                    if let Some(error) = p.iter().find(|arg| arg.is_err()) {
+                        return Err(error.clone().err().unwrap());
+                    }
+                    p.iter().map(|arg| *arg.as_ref().unwrap() as u32).collect_vec()
+                } else {
+                    vec![]
+                };
+
+                if let Some(x) = name {
+                    let constructor = if let Some(cons) = connector.find_native_type_constructor(x) {
+                        cons
+                    } else {
+                        return Err(DatamodelError::new_connector_error(
+                            &ConnectorError::from_kind(ErrorKind::NativeTypeNameUnknown {
+                                native_type: x.parse().unwrap(),
+                                connector_name: connector_string.clone(),
+                            })
+                            .to_string(),
+                            type_specifications.first().unwrap().span,
+                        ));
+                    };
+
+                    let native_type_constructor = connector
```

Comment:
Isn't this exactly the same as the `constructor` variable above?

Reply:
yes it is, will remove it